### ISSUE TITLE
cligen doesn't distinguish bw `a: seq[int]` and `a: seq[string]` (etc)

### ISSUE THREAD

**Issue by timotheecour** [c-blake/cligen#60]
test case: see D20181130T094043 in https://github.com/timotheecour/vitanim/blob/master/testcases/tests/t0000.nim

**Comment by c-blake** [comment]
It could default to the parameter name, like it is now, but maybe with a type in parens, like "a(int)", but then if the user gives me a help["a"] I would use whatever that says.

**Comment by timotheecour** [comment]
yes I was talking about providing better help msg by default; note that ppl are lazy and (most of the time IMO) cligen will be used without a `help` keys; note also that DRY is good so user shouldn't have to specify type in help msg; hence my proposal below:

- [ ] how about following:
```
when true:
  proc main(foo: seq[int])=
    echo foo
dispatch(main)
```
shall produce:
```
Usage:
  main [optional-params] [foo: int...]
  Options(opt-arg sep :|=|spc):
  -h, --help      write this help to stdout
```

- [ ] furthermore, how about making a help keyed on foo generate following instead of what you did in https://github.com/c-blake/cligen/commit/a4490a4810984ffa3bd9090da7445c5815b9dc54 :
```
dispatch(main, help={ "foo" : "process specified inputs" })
```
shall generate:
```
Usage:
  main [optional-params] [foo: int...]
  foo: process specified inputs
  Options(opt-arg sep :|=|spc):
  -h, --help      write this help to stdout
```


## notes
Note that `...` is common in help msgs (with variations on how it's presented), with same meaning as in our case, eg:
```
ls --help
Usage: gls [OPTION]... [FILE]...

man ls
SYNOPSIS
     ls [-ABCFGHLOPRSTUW@abcdefghiklmnopqrstuwx1] [file ...]
```

**Comment by c-blake** [comment]
For my commands where you have `foo` I have `paths` and so it reads `[paths]` which I think is quite clear.  I went ahead and added some "..."s for you, but there's a hook for whatever you want now.

I don't think it's reasonable to (A) be picky about how the help shows up (B) not have good parameter names *and* (C) not use `help`.  So, this seems a really minor issue at best now.  We should document this new `help` thing somewhere/somehow, though.

That said, I wouldn't mind showing the Nim type in there, but I am not sure I have a reliable handle on the `T` inside the `seq[T]` in the macro code.  PR would be welcome on that if you want to try.

About the last thing "processing" thing, I am against it.  The main doc comment for the proc should say whatever it is that it generally does with some input and the parameter names should guide as to what that might be.  If that comment says nothing/doesn't exist and none of the names in play clue in the user, then "process specified `foo`" or whatever generic thing we might say adds absolutely no information.  The arguments might be negative like "skip the things in foo" in which case the generic wording may actually be misleading. Sure, we could say something bureaucratically neutral like "running procName in consideration of procParameter", but yuck.

Best to make people be _slightly_ less lazy or have their users complain about a slightly better help message, I'd say.  Personally, I'd hope anyone writing a command used by others would at least use `help`.

**Comment by c-blake** [comment]
I'm also not against revisiting the "overall help template" formatting engine.  We could have some kind of `$positionalName` for people to interpolate into some standard help format.  Then they could not be lazy just once and `import cligen, myFmt; cligen.template=myCG`.  If someone has a boss who probably incorrectly believes that bureaucratic boilerplate in help messages is _not_ prone to make people skip reading what little other text is there, it could make appeasing such bosses slightly easier.
